.github/actions/nexus-move-artifacts/action.yml

```diff
@@ -0,0 +1,38 @@
+name: Move artifacts to destination repository
+description: Move artifacts to destination repository in Nexus
+inputs:
+  tag-name:
+    description: The name of the tag
+    required: true
+  nexus-username:
+    description: Nexus user name
+    required: true
+  nexus-password:
+    description: Nexus password
+    required: true
+  nexus-url:
+    description: Base URL to the Nexus server
+    required: false
+    default: "https://artifacts.alfresco.com/nexus"
+  destination-repository:
+    description: The destination repository
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: move artifacts
+      id: move artifacts
+      shell: bash
+      env:
+        TAG_NAME: ${{ inputs.tag-name }}
+        NEXUS_USERNAME: ${{ inputs.nexus-username }}
+        NEXUS_PASSWORD: ${{ inputs.nexus-password }}
+        NEXUS_URL: ${{ inputs.nexus-url }}
+        DESTINATION_REPOSITORY: ${{ inputs.destination-repository }}
+      run: |
+        echo "Move artifacts with tag $TAG_NAME to staging repository $STAGING_REPOSITORY"
+        curl -u "${NEXUS_USERNAME}:${NEXUS_PASSWORD}" \
+          -X POST "${NEXUS_URL}/service/rest/v1/tags/move/{DESTINATION_REPOSITORY}?tag=${TAG_NAME}" \
+          -H 'accept: application/json' \
+          -d '{}'
```

Comment:
I think you also need o specify a `source-repository` (`repository` parameter in the API)